### PR TITLE
doc: 2.6.0 doc review

### PR DESCRIPTION
**Pull request by annwoj**
multiple doc changes applied.

### CODE REVIEW

File: doc/setting_up_sidewalk_environment/requirements.rst [hunk 1]
```diff
@@ -49,14 +52,16 @@ All values are provided in kilobytes (KB).
 
 .. tabs::
 
-   .. tab:: `nRF52840 DK`_
+   .. tab:: nRF52840 DK
 
       The following table lists memory requirements for samples running on the `nRF52840 DK`_.
 
+      .. table tbc: Should these be called "end device" given they redirect to the Sidewalk hello variant?
```
Comment (annwoj) [inline doc/setting_up_sidewalk_environment/requirements.rst:59]: TBC (as per comment above)

File: doc/samples/variants/hello.rst [hunk 2]
```diff
@@ -27,34 +37,45 @@ Button 3 (short press):
...
-   Application registered successfully.
+   Indicates that the application registered successfully.
 
 LED 4:
-   Application woke up successfully.
+   Indicates that the application woke up successfully.
 
-Configuration
-*************
+Source file setup
```
Comment (annwoj) [inline doc/samples/variants/hello.rst:52]: Moved it here from the main section. Is it applicable also for DUT?

File: doc/samples/sid_end_device.rst [hunk 3]
```diff
@@ -122,88 +90,59 @@ For example:
...
-      Sidewalk --> [*] : event_factory_reset
-      Sidewalk --> FirmwareUpdate : event_dfu
-      FirmwareUpdate --> Sidewalk : event_dfu
+   variants/hello.rst
+   variants/sensor_monitoring.rst
+   variants/dut.rst
 
 .. _registering_sidewalk:
 
 Registering Sidewalk Endpoint
```
Comment (annwoj) [inline doc/samples/sid_end_device.rst:130]: Probably should've caught that earlier, just making sure my understanding is correct:
Is "registering sidewalk endpoints" applicable to all variants? E.g., in Hello and Sensor monitoring there's a line:
'Wait for the device to complete the automatic registration'. Is it about the same process?
If so then I'm not sure if the current placement of the "registering sidewalk endpoints" is the best. 
In addition, in Hello Sidewalk, in line 120 this is repeated again as
"Wait for the device to register, or perform registration manually as described in the :ref:`registering_sidewalk` documentation."
I think we should unify how and where this is referenced.

File: doc/includes/include_sidewalk_kconfig_options.txt [hunk 4]
```diff
@@ -0,0 +1,25 @@
+* ``CONFIG_SIDEWALK`` -- Enables support for the Sidewalk protocol and its dependencies.
```
Comment (annwoj) [inline doc/includes/include_sidewalk_kconfig_options.txt:1]: Question about the images -> I removed the circles as they didn't have descriptions. As a good practice, the diagram shouldn't have any elements that are not explained. Preferably, I'd like to remove the white circle from "sidewalk_thread_state_machine.svg"  file, though it does have an event that leads to it, so  maybe there is some description that could be addded here?

File: doc/samples/sid_end_device.rst [hunk 5]
```diff
@@ -37,73 +43,51 @@ The supported modules are as follows:
...
-* ``debug`` -- Debug version of the application.
-  It can be used to enable additional features, such as logs or command-line shell, to verify the application behavior.
-* ``release`` -- Release version of the application.
-  It can be used to enable only the necessary application functionalities to optimize its performance.
-* ``no_dfu`` -- No DFU version of the application.
-  It can be used in simple scenarios where DFU is not required.
-  Additionally, it can be treated as a starting point when integrating the Sidewalk sample with a custom bootloader.
+The sample supports the following build types:
+
+.. list-table::
```
Comment (annwoj) [inline doc/samples/sid_end_device.rst:67]: Aligned this section to Matter. Please double-check

**Review by wojciechpluszynski — APPROVED** [review]

**Review by totyz — APPROVED** [review]

**Pull request merged**